plugins/flatpak/gs-flatpak.c

```diff
@@ -2577,6 +2577,43 @@ gs_flatpak_refine_appstream (GsFlatpak *self, GsApp *app, GError **error)
 	return TRUE;
 }
 
+static void
+gs_flatpak_refine_proxy_app (GsFlatpak	*self,
+			     GsApp	*app)
+{
+	GPtrArray *proxied_apps = gs_app_get_related (app);
+	g_autoptr(GsAppList) real_related_apps = NULL;
+
+	if (proxied_apps->len == 0)
+		return;
+
+	real_related_apps = gs_app_list_new ();
+
+	/* get a list of the real apps (not eventual wildcard ones) */
+	for (guint i = 0; i < proxied_apps->len; ++i) {
+		GsApp *app_cached = NULL;
+		GsApp *proxied_app = g_ptr_array_index (proxied_apps, i);
+
+		if (gs_app_has_quirk (proxied_app, GS_APP_QUIRK_IS_WILDCARD) &&
+		    gs_app_get_bundle_kind (proxied_app) == AS_BUNDLE_KIND_FLATPAK) {
```

Comment:
Could we even get here with something that is not a flatpak?

Reply:
Actually we could! I think it could be prevented if we added something in https://github.com/endlessm/gnome-software/blob/master/plugins/flatpak/gs-plugin-flatpak.c#L343
However, maybe the idea was also to have (outside of EOS) proxy apps matching apps in different systems. So let's keep it here for now.